src/solvers/lowering/byte_operators.cpp

```diff
@@ -294,9 +342,12 @@ exprt lower_byte_extract(const byte_extract_exprt &src, const namespacet &ns)
           typecast_exprt(src.offset(), upper_bound.type())),
         ns);
 
+  exprt lb = src.offset();
+  if(!lb.is_constant())
```

Comment:
This constraint suggests passing `optional<constant_exprt>` rather than `exprt`

Reply:
Fair enough - I'll add an extra commit that refactors this.